src/WebJobs.Script.Grpc/MessageExtensions/StatusResultExtensions.cs

```diff
@@ -27,12 +29,18 @@ public static bool IsFailure(this StatusResult statusResult, out Exception excep
             }
         }
 
-        public static bool IsSuccess<T>(this StatusResult status, TaskCompletionSource<T> tcs)
+        public static bool IsInvocationSuccess<T>(this StatusResult status, TaskCompletionSource<T> tcs)
```

Comment:
Having "IsFailure" and "IsInvocationSuccess" together looks inconsistent to me (aka why aren't there corresponding opposites like IsSuccess or IsInvocationFailure, why isn't one parallel structure chosen). 

The difference in wording also implies that these methods are checking different things - IsFailure is checking for any failure, and IsInvocationSuccess is checking just the invocation (but something else may fail?). I see the discussion above about code paths and scenarios covered by these two methods, and from my understanding this naming difference seems consistent with your explanation, but I would add comments to clarify.

Reply:
That's a good point!